packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -824,6 +834,24 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
             message.contents = JSON.parse(message.contents);
         }
 
+        // validate client sequence number has no gap. If there is gap, check if there were noops
+        // if there were noops, decrement the gap by number of noops and continue
+        if (this.connectionManager.lastSubmittedClientId !== undefined && this.connectionManager.lastSubmittedClientId === message.clientId) {
```

Comment:
On a second thought, I think leveraging this.connectionManager.clientId is more correct here (over 
 this.connectionManager.lastSubmittedClientId).

Here is my thinking: once we establish new connection, this.noOpCount will be reset in connectHandler(). But will be pointing to old connection, and we might be still getting ops from that old connection. So IF condition (as it stands right now) will be valid, but this.noOpCount will be already reset, which will cause trouble.

If we change to use this.connectionManager.clientId, then we will lose some validation (during reconnection flow described above), but at least we will not cause fake errors. It's a compromise, but it will give us pretty good (~99%) coverage (i.e. validation through the lifetime of connection, but not through reconnection logic).